app/controllers/application_controller.rb

```diff
@@ -148,8 +154,21 @@ def merge_last_user(attrs)
 
   protected
 
-  def set_entity
-    @entity = Entity.find(params[:id])
+  def set_entity(args = {})
+    @skope = Entity.unscoped.includes(args[:includes])
+    @entity = @skope.find_by_id(params[:id])
+    raise_merged if @entity&.has_merges?
+    raise_if_not_found @entity
+  end
+
+  def raise_merged
```

Comment:
minor point, but these  helpers can be probable be 'private' not 'protected' not that it means that much in ruby....

Reply:
+1

Reply:
( i assume we're just saying let's move `raise_merged` and `raise_if_not_found` to private, but leave `set_entity` public?)